src/Containers/DropArea/DropArea.tsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import styled from 'styled-components';
+import { MainInterface, Main } from '@Utils/BaseStyles';
+import { flex } from '@Utils/Mixins';
+import Dropzone, {
+    useDropzone,
+    DropEvent,
+    FileRejection,
+} from 'react-dropzone';
+
+export interface IDropAreaProps
+    extends MainInterface,
+        React.HTMLAttributes<HTMLDivElement> {
+    message?: string;
+    onClick?: React.MouseEventHandler<HTMLElement>|undefined;
+    onDragEnter?: React.DragEventHandler<HTMLElement> | undefined;
+    onDragLeave?: React.DragEventHandler<HTMLElement> | undefined;
+    onDropHandler?:
+        | (<T extends File>(
+              acceptedFiles: T[],
+              fileRejections: FileRejection[],
+              event: DropEvent,
+          ) => void)
+        | undefined;
+    isDragEnter?: boolean;
+}
+
+export const DropArea: React.FC<IDropAreaProps> = ({
+    message = 'Drag & Drop your files here',
+    onClick=()=>null,
+    onDragEnter = () => null,
+    onDragLeave = () => null,
+    onDropHandler = () => null,
+    isDragEnter = false,
+    ...props
+}): React.ReactElement => {
+    const { getInputProps, getRootProps } = useDropzone({
+        onDragEnter,
+        onDragLeave,
+        onDrop: onDropHandler,
+    });
+    return (
+        <Dropzone multiple>
+            {() => (
```

Comment:
Why is there a function here?

Reply:
because this is render props pattern and is how it works with the Dropzone component.

Reply:
Can't it be abstracted outside the tree?

Reply:
I don't think so, you would be abstracting all the tree

Reply:
Ok let's keep it then I guess